packages/std/src/query/wasm.rs

```diff
@@ -61,3 +64,23 @@ impl ContractInfoResponse {
         }
     }
 }
+
+/// The essential data from wasmd's [CodeInfo]/[CodeInfoResponse].
+///
+/// `code_hash`/`data_hash` was renamed to `checksum` to follow the CosmWasm
+/// convention and naming in `instantiate2_address`.
+///
+/// [CodeInfo]: https://github.com/CosmWasm/wasmd/blob/v0.30.0/proto/cosmwasm/wasm/v1/types.proto#L62-L72
+/// [CodeInfoResponse]: https://github.com/CosmWasm/wasmd/blob/v0.30.0/proto/cosmwasm/wasm/v1/query.proto#L184-L199
+#[non_exhaustive]
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, JsonSchema)]
+#[cfg(feature = "cosmwasm_1_2")]
+pub struct CodeInfoResponse {
+    /// The address that initially stored the code
+    pub creator: String,
+    /// The hash of the Wasm blob
+    pub checksum: Binary,
+}
```

Comment:
Would it make sense to return the `InstantiateConfig` as well? It feels incomplete without

Reply:
I was hesitant to add it since it is a complex deeply nested type that is not trivial to add to all the layers in between. But I can have a second look today.

I agree it feels incomplete. But on the other hand the smaller the interface, the more we can break on the wasmd side without breaking contracts. Do you see a clear use case for this information in the contract?

Reply:
Given that
- there are still things moving around instantiation permissions (https://github.com/CosmWasm/wasmd/issues/1042, https://github.com/CosmWasm/wasmd/issues/1142),
- we currently do not have experience with bringing enums from protobuf via JSON to Rust,
- the type is complex with address fields only being valid in case of certain enum values
I would like to avoid blockin Instantiate2 usage on that. The field can be added later on.